metagraph/src/cli/align.cpp

```diff
@@ -349,10 +348,8 @@ int align_to_graph(Config *config) {
     ThreadPool thread_pool(get_num_threads());
     std::mutex print_mutex;
 
-    if (graph->get_mode() == DeBruijnGraph::PRIMARY) {
-        logger->trace("Primary graph wrapped into canonical");
-        graph = std::make_shared<CanonicalDBG>(graph);
-    }
+    if (graph->get_mode() == DeBruijnGraph::PRIMARY)
+        graph = primary_to_canonical(graph);
```

Comment:
this doesn't make it clear that it will also have a cached view on the graph

Reply:
At this point, DBGSuccinctCached is not used, it's only introduced later when a batch is being aligned.

Reply:
Then how do you explain this signature?
```c++
std::shared_ptr<graph::DeBruijnGraph> primary_to_canonical(std::shared_ptr<Graph> graph,
                                                           size_t cache_size = 1024);
```
I see a non-zero cache size.
Or, is this another argument that is dummy and not used?

Reply:
`CanonicalDBG` and `DBGSuccinctCached` cache different things.

`CanonicalDBG` caches the child and parent nodes of each node (and works for every graph type), whereas `DBGSuccinctCache` caches decoded nodes labels and the BOSS edge of the first nucleotide (when backwards traversal is performed), and the boss edges of the reverse complement (when the reverse complement of a node is computed).

Reply:
Ok, I guess this needs to be clarified. Start with changing the returned type to something more specific instead of the most general `DeBruijnGraph`.

Reply:
`std::shared_ptr<CanonicalDBG<Graph>>`?